src/plugins/python/pythonplugin.cpp

```diff
@@ -18,13 +18,15 @@
  * this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#define Py_LIMITED_API 0x03040000
```

Comment:
I think this explains why you didn't run into the errors I was running into. The whole plugin needs to compile with this define, not only this file! In particular, we need `pythonbind.cpp` to also take this define.

Rather than putting this here, I have added the following line to `python.qbs` (see fa939ef4b4e92a74771ce8426b68ab22fd79856a):

```qbs
cpp.defines: base.concat(["Py_LIMITED_API=0x03040000"])
```

Reply:
Ah, I added the define in the Makefile but guess I never did `make clean` to force it to recompile the bind sources as well.
Now it does give those same errors, will check if there's something that can be done about them.

Reply:
yep it does indeed look like pybindgen would require many changes to generate stable ABI compatible code .. should’ve perhaps read past the PEP384 Abstract and/or realised that just removing one PyRun was way too easy. :)
The biggest hurdle seems to be the new dynamic type objects. There is however a tool that can automate the conversion to some extent http://svn.python.org/projects/python/branches/pep-0384/Tools/scripts/abitype.py ..I’ll give that a go some day next week when there’s a free moment.